Alt Text Edits - Android (#3328)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/android/analytics/setting_custom_attributes.md

---

# Setting custom attributes for Android and FireOS

Braze provides methods for assigning attributes to users. You'll be able to filter and segment your users according to these attributes on the dashboard.

Before implementation, be sure to review examples of the segmentation options afforded by custom events, custom attributes, and purchase events in our [analytics overview][7], as well as our notes on [event naming conventions]({{site.baseurl}}/user_guide/data_and_analytics/custom_data/event_naming_conventions/).

## Assigning default user attributes

To assign attributes to your users, call the `getCurrentUser()` method on your Braze instance to get a reference to the current user of your app. Once you have a reference to the current user, you can call methods to set predefined or custom attributes.

> Braze provides predefined methods for setting the following user attributes within the [BrazeUser class][2]. Refer to our [KDoc for method specifications][2]:

Reviewer: ```suggestion
Braze provides predefined methods for setting the following user attributes within the [BrazeUser class][2]. Refer to our KDoc for [method specifications][2]:
```

- Floats
- Doubles

Custom attribute keys and values have a maximum length of 255 characters. Longer strings will be truncated to 255 characters.

> Full method specifications for custom attributes can be found here within the [BrazeUser class in our KDoc][2].

Reviewer: ```suggestion
Full method specifications for custom attributes can be found here within the [BrazeUser class][2] in our KDoc.
```

### Custom attribute via the REST API

You can also use our REST API to set user attributes. To do so, refer to the [User API documentation][4].

## Setting up user subscriptions

To set up a subscription for your users (either email or push), call the functions `setEmailNotificationSubscriptionType()`  or `setPushNotificationSubscriptionType()`, respectively. Both of these functions take the enum type `NotificationSubscriptionType` as arguments. This type has three different states:

| Subscription status | Definition |
| ------------------- | ---------- |
| `OPTED_IN` | Subscribed, and explicitly opted in |
| `SUBSCRIBED` | Subscribed, but not explicitly opted in |
| `UNSUBSCRIBED` | Unsubscribed and/or explicitly opted out |
{: .reset-td-br-1 .reset-td-br-2}

{% alert important %}

> No explicit opt-in is required by Android to send users push notifications. When a user is registered for push, they are set to `SUBSCRIBED` rather than `OPTED_IN` by default. Visit our [User Guide]({{site.baseurl}}/user_guide/message_building_by_channel/email/managing_user_subscriptions/#managing-user-subscriptions) for more information on implementing subscriptions and explicit opt-ins.

Reviewer: With "User Guide" as the link text, I'd expect the link to take me to the user guide home page. Suggesting the article title instead! :)

```suggestion
No explicit opt-in is required by Android to send users push notifications. When a user is registered for push, they are set to `SUBSCRIBED` rather than `OPTED_IN` by default. Refer to [managing user subscriptions]({{site.baseurl}}/user_guide/message_building_by_channel/email/managing_user_subscriptions/#managing-user-subscriptions) for more information on implementing subscriptions and explicit opt-ins.
```
